src/api/java/com/minecolonies/api/colony/buildings/modules/IBuildingModule.java

```diff
@@ -0,0 +1,109 @@
+package com.minecolonies.api.colony.buildings.modules;
+
+import com.minecolonies.api.colony.IColony;
+import com.minecolonies.api.colony.buildings.IBuilding;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.network.PacketBuffer;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Interface describing the different modules.
+ */
+public interface IBuildingModule
+{
+    /**
+     * Deserialize the module.
+     * @param compound the nbt compound.
+     */
+    default void deserializeNBT(CompoundNBT compound)
+    {
+
+    }
+
+    /**
+     * Serialize the module from a compound.
+     * @param compound the compound.
+     */
+    default void serializeNBT(final CompoundNBT compound)
+    {
+
+    }
+
+    /**
+     * Serialization method to send the module data to the client side.
+     * @param buf the buffer to write it to.
+     */
+    default void serializeToView(PacketBuffer buf)
```

Comment:
We should declare a standard interface for serialization to and from `PacketBuffer` and treat it in the same way vanilla treats NBT with its `NBTSerializable` interface. Aka create an empty instance and then call deserialize on it.

I am pretty sure this interface was added when we added the PacketBuffer specific serialization for the RS.
This also makes it so that Views are not needed for modules, which is a huge improvement in maintainability.

Reply:
As I wrote above, views will be a part of modules. In a way that allows us to build nice GUIs from multiple module Views